### PR TITLE
fix(suggested-fees): Quote timestamp handling

### PR DESCRIPTION
**Pull request by pxrl**
- Never quote on future timestamps.
- If the user supplied a timestamp, don't modify it.
- Optimise by resolving the current block simultaneously with the token
  details.

Closes ACX-1481

### CODE REVIEW

File: api/suggested-fees.ts [hunk 1]
```diff
@@ -72,39 +72,41 @@ const handler = async (
...
+      parsedTimestamp =
+        Math.floor((latestBlock.timestamp - quoteTimeBuffer) / precision) *
+        precision;
+    }
 
-    const amount = ethers.BigNumber.from(amountInput);
+    // Don't attempt to provide quotes for future timestamps.
+    if (parsedTimestamp > latestBlock.timestamp) {
+      throw new InputError("Invalid quote timestamp");
+    }
```
Comment (james-a-morris) [inline api/suggested-fees.ts:107]: Good call

File: api/suggested-fees.ts [hunk 2]
```diff
@@ -72,39 +72,41 @@ const handler = async (
...
+    ]);
+    const { l1Token, hubPool, chainId: computedOriginChainId } = tokenDetails;
+
+    // Note: Add a buffer to "latest" timestamp so that it corresponds to a block older than HEAD.
+    // This is to improve relayer UX who have heightened risk of sending inadvertent invalid fills
+    // for quote times right at HEAD (or worst, in the future of HEAD). If timestamp is supplied as
+    // a query param, then no need to apply buffer.
+
+    // If the caller did not supply a quote timestamp, generate one from the latest block number.
+    let parsedTimestamp = Number(timestamp);
```
Comment (james-a-morris) [inline api/suggested-fees.ts:87]: Can't this be `const`?

**Review by james-a-morris — COMMENTED** [review]
Left some comments

File: api/suggested-fees.ts (hunk 2 above)
Comment (pxrl) [reply api/suggested-fees.ts:87]: It can be re-set on line 99 (below).

**Review by nicholaspai — APPROVED** [review]

**Review by dohaki — APPROVED** [review]

**Review by dohaki — APPROVED** [review]

File: api/suggested-fees.ts [hunk 3]
```diff
@@ -72,39 +72,41 @@ const handler = async (
...
+    if (isNaN(parsedTimestamp)) {
```
[the remark this answers is not in the document]
Comment (pxrl) [reply api/suggested-fees.ts:88]: Seems like we already leave the timestamp undefined: https://github.com/across-protocol/frontend-v2/blob/6d61b703459614ce6aa2a065bef5d36fd52f1aa7/src/utils/serverless-api/prod/suggested-fees.prod.ts#L20.

**Pull request merged**